### PR TITLE
Add get options cf method for DB

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR adds the method `get_options_cf` for `DB`.

### CODE REVIEW

**Comment by hhkbp2** [comment]
PTAL @zhangjinpeng1987 @siddontang @BusyJay

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -1053,6 +1053,19 @@ void crocksdb_enable_file_deletions(
...
+crocksdb_options_t* crocksdb_get_options(const crocksdb_t* db) {
+  crocksdb_options_t* options = new crocksdb_options_t;
+  options->rep = db->rep->GetOptions();
+  return options;
+}
+
+crocksdb_options_t* crocksdb_get_options_cf(
+    const crocksdb_t* db, crocksdb_column_family_handle_t* column_family) {
+  crocksdb_options_t* options = new crocksdb_options_t;
+  options-> rep = db->rep->GetOptions();
```
Comment (zhangjinpeng1987) [inline librocksdb_sys/crocksdb/c.cc]: Please format code.

Comment (zhangjinpeng1987) [inline librocksdb_sys/crocksdb/c.cc]: forget column_family?

File: src/rocksdb.rs [hunk 2]
```diff
@@ -920,6 +920,14 @@ impl DB {
         &self.opts
     }
 
+    pub fn get_options_cf(&self, cf: &CFHandle) -> Options {
+        unsafe {
+            let inner = crocksdb_ffi::crocksdb_get_options_cf(self.inner, cf.inner);
+            assert!(!inner.is_null(), "could not get options cf");
```
Comment (zhangjinpeng1987) [inline src/rocksdb.rs]: inner never be null.

Comment (zhangjinpeng1987) [inline src/rocksdb.rs]: duplicated assert at here and in `new_with`

**Comment by siddontang** [comment]
we can remove https://github.com/pingcap/rust-rocksdb/blob/82daf985debe78a46aa6761e3895bb6774457f77/src/rocksdb.rs#L52 and all use rocksdb API.

File: src/rocksdb.rs [hunk 3]
```diff
@@ -920,6 +920,13 @@ impl DB {
         &self.opts
     }
 
+    pub fn get_options_cf(&self, cf: &CFHandle) -> Options {
```
Comment (siddontang) [inline src/rocksdb.rs]: Use `crocksdb_ffi::crocksdb_get_options` for `get_options` too.

File: librocksdb_sys/crocksdb/c.cc [hunk 4]
```diff
@@ -1053,6 +1053,20 @@ void crocksdb_enable_file_deletions(
...
+  crocksdb_options_t* options = new crocksdb_options_t;
+  options->rep = db->rep->GetOptions();
+  return options;
+}
+
+crocksdb_options_t* crocksdb_get_options_cf(
+    const crocksdb_t* db,
+    crocksdb_column_family_handle_t* column_family) {
+  crocksdb_options_t* options = new crocksdb_options_t;
+  options-> rep = db->rep->GetOptions(column_family->rep);
```
Comment (zhangjinpeng1987) [inline librocksdb_sys/crocksdb/c.cc]: s/options-> rep/options->rep/

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by hhkbp2** [comment]
@siddontang 
https://github.com/pingcap/rust-rocksdb/blob/82daf985debe78a46aa6761e3895bb6774457f77/src/rocksdb.rs#L52

This `opts` should not be removed since there are function pointers in `opts`, which must be valid as long as the db are opened.

**Comment by siddontang** [comment]
Got it. please add opts again.

**Comment by hhkbp2** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
LGTM

File: src/rocksdb_options.rs [hunk 5]
```diff
@@ -285,6 +285,15 @@ impl Options {
         Options::default()
     }
 
+    pub fn new_with(inner: *mut DBOptions) -> Options {
```
Comment (BusyJay) [inline src/rocksdb_options.rs]: Should mark as unsafe.

File: tests/test_ingest_external_file.rs [hunk 6]
```diff
@@ -43,7 +43,8 @@ fn test_ingest_external_file() {
     let test_sstfile = gen_path.path().join("test_sst_file");
     let test_sstfile_str = test_sstfile.to_str().unwrap();
 
-    gen_sst(db.get_options(),
+    let default_options = db.get_options();
+    gen_sst(&default_options,
```
Comment (BusyJay) [inline tests/test_ingest_external_file.rs]: You can change the signature of `gen_sst` directly.

**Pull request merged**